tests/include/helpers.h

```diff
@@ -69,6 +69,382 @@ typedef UINT32 uint32_t;
 #include <strings.h>
 #endif
 
+/*----------------------------------------------------------------------------*/
+/* Helper flags for complex dependencies */
+
+/* Indicates whether we expect mbedtls_entropy_init
+ * to initialize some strong entropy source. */
+#if defined(MBEDTLS_TEST_NULL_ENTROPY) ||             \
+    ( !defined(MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES) && \
+      ( !defined(MBEDTLS_NO_PLATFORM_ENTROPY)  ||     \
+         defined(MBEDTLS_HAVEGE_C)             ||     \
+         defined(MBEDTLS_ENTROPY_HARDWARE_ALT) ||     \
+         defined(ENTROPY_NV_SEED) ) )
+#define ENTROPY_HAVE_STRONG
+#endif
+
+/*----------------------------------------------------------------------------*/
+/* Status and error constants */
```

Comment:
The dispatch status values are an internal detail of the toplevel loop of the test code, not for use in test code. Since this PR is precisely about moving code around, I think it would be better to move this directly to a separate header file (even if this header would currently be included in the same places). But this isn't a blocker.

This also applies to the definition of `test_info_t` and related types.

Reply:
I don't move this good in the new version of the PR.